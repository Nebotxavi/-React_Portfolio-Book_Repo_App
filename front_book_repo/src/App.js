import React, { useState, useEffect } from "react";
import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import "font-awesome/css/font-awesome.css";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

import { Route, Switch, Redirect } from "react-router-dom";

import Books from "./components/books";
import BookForm from "./components/bookForm";
import { getCurrentUser } from "./services/authService";
import LoginForm from "./components/loginForm";
import Logout from "./components/logout";
import NavBar from "./components/navBar";
import NotFound from "./components/common/notFound";
import ProtectedRoute from "./components/common/protectedRoute";
import RegisterForm from "./components/registerForm";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

function App() {
  const [user, setUser] = useState();

  useEffect(() => {
    const username = getCurrentUser();
    setUser(username);
  }, []);

  return (
    <React.Fragment>
      <ToastContainer />
      <NavBar user={user} />
      <div className="content">
        <Switch>
          <ProtectedRoute path="/book/:id" component={BookForm} />
          <Route
            path="/books"
            render={props => <Books {...props} user={user} />}
          />
          <Route path="/login" component={LoginForm} />
          <Route path="/logout" component={Logout} />
          <Route path="/register" component={RegisterForm} />
          <Redirect from="/" exact to="books" />
          <Route path="/not-found" component={NotFound} />
          <Redirect to="/not-found" />
        </Switch>
      </div>
    </React.Fragment>
  );
}

export default App;
