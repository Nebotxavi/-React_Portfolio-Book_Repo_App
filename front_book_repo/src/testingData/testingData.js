export const books = [
  {
    id: 1,
    title: "Cold Blood",
    author_first_name: "Truman",
    author_second_name: "Capote",
    genre: 1,
    rate: 10
  },
  {
    id: 2,
    title: "En la frontera",
    author_first_name: "Cormac",
    author_second_name: "McCarthy",
    genre: 1,
    rate: 10
  },
  {
    id: 3,
    title: "Testing 1",
    author_first_name: "Pep",
    author_second_name: "Atlas",
    genre: 2,
    rate: 6
  },
  {
    id: 4,
    title: "Peim 1",
    author_first_name: "Pep",
    author_second_name: "Nyony",
    genre: 2,
    rate: 7
  }
];

export const genres = [
  {
    name: "americana"
  },
  {
    name: "drama"
  }
];
