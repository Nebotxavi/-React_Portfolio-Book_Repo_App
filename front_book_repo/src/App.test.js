import React from "react";
import { render } from "@testing-library/react";

import Adapter from "enzyme-adapter-react-16";
import { BrowserRouter } from "react-router-dom";
import { configure, mount, shallow } from "enzyme";
import renderer from "react-test-renderer";

import App from "./App";
import BooksTable from "./components/booksTable";
import { books, genres } from "./testingData/testingData";
import FilterMenu from "./components/common/filterMenu";
import NavBar from "./components/navBar";

// Enzyme setup
configure({ adapter: new Adapter() });

// Snapshot testing

describe("NavBar", () => {
  test("NavBar component renders (snapshot)", () => {
    const component = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Books table", () => {
  test("Books table component renders (snapshot)", () => {
    const component = renderer.create(<BooksTable books={books} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Filter menu", () => {
  test("Filter menu component renders (snapshot)", () => {
    const component = renderer.create(<FilterMenu items={genres} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// AXIOS TESTING
