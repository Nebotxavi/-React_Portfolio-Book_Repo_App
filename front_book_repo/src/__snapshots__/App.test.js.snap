// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books table Books table component renders (snapshot) 1`] = `
<table
  className="table table-hover"
>
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Genre
      </th>
      <th>
        Rate
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Cold Blood
      </td>
      <td>
        Truman Capote
      </td>
      <td>
        1
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        En la frontera
      </td>
      <td>
        Cormac McCarthy
      </td>
      <td>
        1
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        Testing 1
      </td>
      <td>
        Pep Atlas
      </td>
      <td>
        2
      </td>
      <td>
        6
      </td>
    </tr>
    <tr>
      <td>
        Peim 1
      </td>
      <td>
        Pep Nyony
      </td>
      <td>
        2
      </td>
      <td>
        7
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Filter menu Filter menu component renders (snapshot) 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item list-group-item-action disabled"
  >
    americana
  </li>
  <li
    className="list-group-item list-group-item-action disabled"
  >
    drama
  </li>
</ul>
`;

exports[`NavBar NavBar component renders (snapshot) 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Book Repo
  </a>
  <button
    aria-controls="navbarNavAltMarkup"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNavAltMarkup"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNavAltMarkup"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/books"
        onClick={[Function]}
      >
        Books
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</nav>
`;
